### PR TITLE
fix: k8s > 1.24 do not support autoscaling/v2beta1

### PR DESCRIPTION
**Pull request by jonathanwiemers**
#### Summary
`autoscaling/v2beta1` is not longer supported in K8S >= 1.25

### CODE REVIEW

**Review by PK89SC — APPROVED** [review]

File: charts/mattermost-enterprise-edition/templates/deployment-mattermost-hpa.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.mattermostApp.autoscaling.enabled }}
-apiVersion: autoscaling/v2beta1
+apiVersion: autoscaling/v2
```
Comment (spirosoik) [inline charts/mattermost-enterprise-edition/templates/deployment-mattermost-hpa.yaml:2]: We need a conditional check here. For example:
```
{{- if semverCompare ">= 1.25-0" .Capabilities.KubeVersion.Version }}
apiVersion: autoscaling/v2
{{ else }}
apiVersion: autoscaling/v2beta1
```

With this ways it's backward compatible.

Comment (jonathanwiemers) [reply charts/mattermost-enterprise-edition/templates/deployment-mattermost-hpa.yaml:3]: @spirosoik updated it

**Comment by jonathanwiemers** [comment]
@spirosoik pls re-review

File: charts/mattermost-enterprise-edition/templates/deployment-mattermost-hpa.yaml (hunk 1 above)
Comment (spirosoik) [reply charts/mattermost-enterprise-edition/templates/deployment-mattermost-hpa.yaml:3]: awesome. Thanks @jonathanwiemers

**Review by spirosoik — APPROVED** [review]
Thanks for the contribution :)

**Pull request merged**